persistence.js

```diff
@@ -219,69 +217,37 @@ function returnSubsForClient (subs) {
 
 RedisPersistence.prototype.countOffline = function (cb) {
   var that = this
-
   this._db.scard(clientsKey, function countOfflineClients (err, count) {
     if (err) {
       return cb(err)
     }
 
-    cb(null, that._trie.subscriptionsCount, parseInt(count) || 0)
-  })
-}
-
-RedisPersistence.prototype.subscriptionsByTopic = function (topic, cb) {
-  if (!this.ready) {
-    this.once('ready', this.subscriptionsByTopic.bind(this, topic, cb))
-    return this
-  }
-
-  var result = this._trie.match(topic)
-
-  cb(null, result)
-}
-
-RedisPersistence.prototype._setup = function () {
-  if (this.ready) {
-    return
-  }
-
-  var that = this
+    that._db.get(subscriberCountKey, function subCount (err, subCount) {
+      if (err) {
+        return cb(err)
+      }
 
-  var hgetallStream = throughv.obj(function getStream (clientId, enc, cb) {
-    var clientSubKey = clientKey + clientId
-    that._db.hgetall(clientSubKey, function clientHash (err, hash) {
-      cb(err, { clientHash: hash, clientId: clientId })
+      cb(null, parseInt(subCount) || 0, parseInt(count) || 0)
     })
-  }, function emitReady (cb) {
-    that.ready = true
-    that.emit('ready')
-    cb()
-  }).on('data', function processKeys (data) {
-    processKeysForClient(data.clientId, data.clientHash, that)
   })
+}
 
-  this._db.smembers(clientsKey, function smembers (err, clientIds) {
+RedisPersistence.prototype.subscriptionsByTopic = function (topic, cb) {
+  const subKey = subscribersKey + topic
+  this._db.hgetall(subKey, function clientHash (err, data) {
     if (err) {
-      hgetallStream.emit('error', err)
-    } else {
-      for (var i = 0, l = clientIds.length; i < l; i++) {
-        hgetallStream.write(clientIds[i])
-      }
-      hgetallStream.end()
+      return cb(err)
```

Comment:
Maybe `cb(err, null)` in case someone decides to use the result and/or compare it to `null` directly for whatever weird reason? Just being cautious here :)

Reply:
👍